packages/web/src/components/premium-content-purchase-modal/components/PurchaseDetailsPage.tsx

```diff
@@ -35,11 +48,14 @@ export const PurchaseDetailsPage = ({
 }) => {
   const dispatch = useDispatch()
   const stage = useSelector(getPurchaseContentFlowStage)
-  const isUnlocking = [
-    PurchaseContentStage.BUY_USDC,
-    PurchaseContentStage.PURCHASING,
-    PurchaseContentStage.CONFIRMING_PURCHASE
-  ].includes(stage)
+  const error = useSelector(getPurchaseContentError)
+  const isUnlocking =
+    !error &&
+    [
+      PurchaseContentStage.BUY_USDC,
+      PurchaseContentStage.PURCHASING,
+      PurchaseContentStage.CONFIRMING_PURCHASE
+    ].includes(stage)
```

Comment:
maybe a good candidate for a shared fn in common?

Reply:
Oh yeah good call. I can move it.